traffic_ops/traffic_ops_golang/deliveryservice/deliveryservicesv13.go

```diff
@@ -959,6 +959,7 @@ ds.ccr_dns_ttl,
 ds.cdn_id,
 cdn.name as cdnName,
 ds.check_path,
+ds.consistent_hash_regex,
```

Comment:
I'm not totally sure we can add fields to API v1.3

Reply:
We can't. Per Semantic Versioning, which we've agreed to as a project, it needs to be a new minor version. But, right now it's a pain, it's a lot of boilerplate that has to be copy-pasted for a new version.

It's ok for `master` not to be perfect, as long I think I'm inclined to Approve this, with the caveat that we fix it before the next release. I'm willing to commit to fixing it; but, I'm also kind of hoping to propose a better solution.

Reply:
idk how much of a pain this is, I think you'd need to rewrite a bit of the v13 handling code and then totally re-implement it for a different API version (again, no idea how much work that is), but from a struct point of view this should be as easy as `type DeliveryServiceNullable struct {` -> `type DeliveryServiceNullableV13 struct {` and

```go
type DeliveryServiceNullable struct {
    DeliveryServiceNullableV13
    consistent_hash_regex *string `json:"consistentHashRegex,omitempty"`
}
```

Reply:
>from a struct point of view this should be as easy as `type DeliveryServiceNullable struct {` -> `type DeliveryServiceNullableV13 struct {` and

Correct.

>I think you'd need to rewrite a bit of the v13 handling code and then totally re-implement it for a different API version

Also correct. It's less rewriting, and more copy-paste. You basically copy the v12 file, name it v13, and rename v13 to v15. 

>no idea how much work that is

It's not really much work, it'd probably take me 15mins. But it'd take longer for someone who didn't already know how. It's mostly just the copy-paste, which people also (rightly) object to.

This is what I meant on the other comment, about metaprogramming. We really need some form of MP—generics, macros, anything would do—and Go simply doesn't have it. The "idiomatic" solution is `go generate`, which, generating code can be viewed as a form of metaprogramming. But people object to that too, just because it's idiomatic doesn't mean it's good.

Again, I'm planning to propose better solution, and I also know what needs done as-is, so I'm willing to commit to fixing this afterwards.

FWIW I'm also planning to create dev docs, either the existing or new solution, for how to add new versions. Hopefully that will help, too.

Reply:
It could help to pass the API version into version-independent handlers for each endpoint under `/api/1.x/*`, then adding a new route can be as easy as adding a `case` in a few places for selective inclusion of fields.